### ISSUE TITLE
Add FORD documentation for Fortran sources

### ISSUE THREAD

**Issue by zbeekman** [sourceryinstitute/opencoarrays#102]
At the very least `opencoarrays.F90` should be documented. Test should probably be documented as well as some build system stuff etc.

**Comment by rouson** [comment]
Agreed.  If you get started, I'll contribute via the time-honored copy-paste-modify method.

**Comment by zbeekman** [comment]
@rouson What's the status of the stuff in `doc/`? Is any of it needed? I was going to output the FORD documentation there, but by default it nukes everything in the output directory... If that stuff is not needed, we should remove it.

**Comment by rouson** [comment]
@zbeekman Given that we're adopting FORD, everything in [doc/](https://github.com/sourceryinstitute/opencoarrays/blob/master/doc/) can be deleted _except_ for [doc/dependency-tree/](https://github.com/sourceryinstitute/opencoarrays/tree/master/doc/dependency_tree), which should be retained. The banner that install.sh prints at the beginning of execution contains the output of `tree doc/dependency-tree` if `tree` is installed and in the user's path.  Otherwise, it prints a static version of the street stored in [doc/dependency_tree/opencoarrays-tree.txt](https://github.com/sourceryinstitute/opencoarrays/blob/master/doc/dependency_tree/opencoarrays-tree.txt).

**Comment by zbeekman** [comment]
OK... Given that FORD nukes the entire doc tree every time it rebuilds, can we relocate this to somewhere like `prerequisites`?

**Comment by rouson** [comment]
Sure.  Feel free to do it or let me know if you'd like for me to do it.

**Comment by zbeekman** [comment]
I'm taking care of it...

**Comment by zbeekman** [comment]
@rouson The one major impediment right now is the `include "compiler_capabilities.txt"` in the integration tests...

**Comment by zbeekman** [comment]
@rouson Uploaded first draft of API documentation to https://sourceryinstitute.github.io/opencoarrays !

Please note that I have yet to add any documentation to any of the source files, everything you get at the link above is generated by FORD, just by adding a "project file".

Also, FORD may not suite our needs for the C/header documentation ( #82 ) but we can start writing it in such a way that we could very easily transition to Doxygen for those portions of the code if we wanted to...

**Comment by rouson** [comment]
Bravo!  :clap:

**Comment by zbeekman** [comment]
Added logic to top-level CMakeLists.txt to build documentation. Still haven't added logic to install it into a proper location. It will fail gracefully (i.e. the build can proceed) if FORD is not installed on the user's system.
